src/routes/auth/providers/github.ts

```diff
@@ -0,0 +1,122 @@
+import { Router, Request, Response } from 'express'
+import passport from 'passport'
+import passportGitHub, { Profile } from 'passport-github2'
+import Boom from '@hapi/boom'
+import { HasuraAccountProviderData } from '@shared/helpers'
+import { selectAccountProvider, insertAccount } from '@shared/queries'
+import { request } from '@shared/request'
+
+import {
+  // PROVIDERS_SUCCESS_REDIRECT,
+  PROVIDERS_FAILURE_REDIRECT,
+  AUTH_GITHUB_CLIENT_ID,
+  AUTH_GITHUB_CLIENT_SECRET,
+  AUTH_GITHUB_CALLBACK_URL,
+  AUTH_GITHUB_AUTHORIZATION_URL,
+  AUTH_GITHUB_TOKEN_URL,
+  AUTH_GITHUB_USER_PROFILE_URL
+} from '@shared/config'
+
+const router = Router()
+const GitHubStrategy = passportGitHub.Strategy
+
+passport.use(
+  new GitHubStrategy(
+    {
+      clientID: AUTH_GITHUB_CLIENT_ID as 'string',
+      clientSecret: AUTH_GITHUB_CLIENT_SECRET as 'string',
+      callbackURL: AUTH_GITHUB_CALLBACK_URL as 'string',
+      authorizationURL: AUTH_GITHUB_AUTHORIZATION_URL as 'string',
+      tokenURL: AUTH_GITHUB_TOKEN_URL as 'string',
+      userProfileURL: AUTH_GITHUB_USER_PROFILE_URL as 'string',
+      scope: ['user:email']
+    },
+    async (
+      accessToken: string,
+      refreshToken: string,
+      profile: Profile,
+      done: (error: any, user?: any) => void
```

Comment:
@plmercereau How could I handle this better?

I am using this: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport-github2/index.d.ts

Reply:
@elitan I just push an improvement, I think we can do better, working on it

Reply:
Nah, even the [typings tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/passport-github2/passport-github2-tests.ts) are not better... Let's park an additional improvement, it's already ok I think

Reply:
Got it. Thanks for taking a look.